### PR TITLE
Add bundle size reporting to PRs

### LINKED ISSUE #567: Add bundle size reporting to PRs
## Problem

Currently if we add or update a dependency, we don't have any automatic insight into how it effects our bundle sizes. Especially when adding a new dependency, this can be helpful in narrowing down options or debugging sudden unexpected increases in bundle sizes.

## Description

Let's add a bundle size analysis step to the CI similar to how Gutenberg has it. [Here is an example comment from a PR where this feedback was helpful for avoiding a sudden unexpected increase in bundle sizes](https://github.com/WordPress/gutenberg/pull/27184#issuecomment-731874775).

We can copy the strategy used by [Gutenberg in this workflow action](https://github.com/WordPress/gutenberg/blob/f06eaf26811a8a8c01602ca8091bca1c302fcb86/.github/workflows/bundle-size.yml#L1). 

The [preactjs@compressed-size](preactjs/compressed-size-action) action should still work fine for us even though we're not using preact or React. It just does a Webpack bundle analysis to get the data.

If for whatever it doesn't work for us then we'll need to explore a bespoke strategy. I tried searching around for `nuxt` specific solutions but there don't appear to be any but we could build off of `nuxt build --analyze --spa --no-generate` and use the generated json stats output in the `./.nuxt/stats` folder.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #567 by @sarayourfriend 

## Description

Uses the example from Gutenberg and uses the preact action used there as well. Hopefully it works :grin: 

Update: it works!

Note that the list of files in this PR is going to be super long because I've also changed how the bundle names are generated.

Maybe we could actually only change the way the bundle names are generated during this specific build by passing an environment variable or something. What do y'all think about that? Then we wouldn't have to worry about hashes changing at all, we'd just see names 1:1? I'll try it out and see if it works.

## Testing Instructions

Check out the new workflow and confirm that it is working :tada: 

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [ ] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by sarayourfriend** [comment]
> Total Size: 0 B

Hmmm, that doesn't seem right :frowning_face:

**Comment by sarayourfriend** [comment]
I think it looks good with the file names for each bundle instead of the hashes. What do y'all think? Theoretically this shouldn't change the build output at all!

File: nuxt.config.ts [hunk 1]
```diff
@@ -108,6 +108,26 @@ const head = {
   ],
 }
 
+const baseProdName = process.env.CI ? '[name]' : '[contenthash:7]'
+
+const filenames: NonNullable<NuxtConfig['build']>['filenames'] = {
```
Comment (obulat) [inline nuxt.config.ts:113]: Awesome! It is definitely much clearer this way!

**Review by obulat — APPROVED** [review]
Thank you for adding this, it's really important!

**Comment by sarayourfriend** [comment]
@dhruvkb do you have time to review this this week? It'd be a nice improvement to devex if we could start tracking this, especially when adding or removing dependencies.

**Review by dhruvkb — APPROVED** [review]
I didn't realise it was ready for a review, the PR description mentioned some things you were planning to try. Nonetheless, this is great and will be very helpful to track changes to our bundle sizes.

**Comment by sarayourfriend** [comment]
Ah sorry, subsequent comments added details about that and I never updated the PR description.

**Pull request merged**